fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/GenericHandler.java

```diff
@@ -107,11 +109,14 @@ public static void handleCatchup(
               + id);
     }
 
-    Log.d(TAG, "messages length: " + messages.size());
+    Log.d(TAG, "Messages to handle : " + messages.size());
     // Handle all received messages from the catchup
     for (MessageGeneral msg : messages) {
-      boolean enqueue = handleMessage(laoRepository, channel, msg);
-      if (enqueue) {
+      try {
+        handleMessage(laoRepository, channel, msg);
+      } catch (DataHandlingException ex) {
+        Log.e(TAG, "Unable to handle message", ex);
+        // This will have to change at some point
```

Comment:
Probably true for any line of code in any program ;-)
Can we have more insight why you think so ?

Reply:
I updated it